### PR TITLE
Containerized proxy services installed in 4.3-*, uyuni-* and head test environments

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Containerized proxy services installed in 4.3-*, uyuni-* and head test environments

### CODE REVIEW

File: salt/repos/server.sls [hunk 1]
```diff
@@ -172,7 +172,7 @@ server_pool_repo:
     - priority: 97
 {% endif %}
 
-{% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') %}
+{% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') or 'uyuni-pr' in grains.get('product_version') %}
```
Comment (srbarrios) [inline salt/repos/server.sls:175]: @rjmateus @cbosdo Should we also add the repo for 4.3 case here?

```
{% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') or 'uyuni-pr' in grains.get('product_version') %}
{% if grains['osfullname'] == 'Leap' %}
proxy_pool_repo:
  pkgrepo.managed:
    - baseurl: http://{{ grains.get("mirror") | default("download.opensuse.org", true) }}/repositories/systemsmanagement:/Uyuni:/Master/images/repo/Uyuni-Proxy-POOL-x86_64-Media1/
    - refresh: True
    - priority: 97
{% else %}
proxy_pool_repo:
  pkgrepo.managed:
    - baseurl: http://{{ grains.get("mirror") | default("download.suse.de", true) }}/ibs/Devel:/Galaxy:/Manager:/Head/images/repo/SLE-Module-SUSE-Manager-Proxy-4.3-POOL-x86_64-Media1/
    - refresh: True
    - priority: 97
{% endif %}
```

Comment (cbosdo) [reply salt/repos/server.sls:175]: Not on the server for sure: the `uyuni-proxy-systemd-services` package is only to be installed on the proxy host

File: salt/proxy/init.sls [hunk 2]
```diff
@@ -40,10 +40,16 @@ proxy-packages:
...
 {% if install_proxy_container_packages %}
 
-{% if 'head' in grains.get('product_version') %}
-    {% set client_tools_repo =  grains.get("mirror") | default("download.suse.de/ibs", true) ~ '/Devel:/Galaxy:/Manager:/Head:/SLE15-SUSE-Manager-Tools/images/repo/SLE-15-Manager-Tools-POOL-x86_64-Media1/' %}
-{% else %}
+{% if 'uyuni' in grains.get('product_version') %}
     {% set client_tools_repo =  grains.get("mirror") | default("download.opensuse.org", true) ~ '/repositories/systemsmanagement:/Uyuni:/Master:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/' %}
+{% elif '4.3' in grains.get('product_version') %}
+  {% if '-released' in grains.get('product_version') %}
+    {% set client_tools_repo = grains.get("mirror") | default("download.suse.de", true) ~ '/ibs/SUSE/Updates/SLE-Manager-Tools/15/x86_64/update' %}
```
Comment (srbarrios) [inline salt/proxy/init.sls:47]: `http://minima-mirror.mgr.suse.de/ibs/SUSE/Updates/SLE-Manager-Tools/15/x86_64/update`

It doesn't seems to work, we will need to update the mirrors

**Review by srbarrios — COMMENTED** [review]
There is only one repo that seems not mirrored, that I would like to have it working before we merge. Aside, I can't find any more issue. So I hope it's good :)

File: salt/proxy/init.sls (hunk 2 above)
Comment (srbarrios) [reply salt/proxy/init.sls:47]: Could you change your URI by that one please?

```suggestion
    {% set client_tools_repo = grains.get("mirror") | default("download.suse.de", true) ~ '/SUSE/Updates/SLE-Manager-Tools/15/x86_64/update' %}
```

Comment (cbosdo) [reply salt/proxy/init.sls]: Fixed. Almost like your suggestion, but the `ibs` part is still needed for `download.suse.de`

**Pull request merged**